agent/fw_cakephp.c

```diff
@@ -193,37 +76,24 @@ NR_PHP_WRAPPER(nr_cakephp_name_the_wt_2) {
     }
   }
 
-  arg1 = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
-  if (!nr_php_is_zval_valid_object(arg1)) {
-    NR_PHP_WRAPPER_CALL;
-    goto end;
-  }
-
   NR_PHP_WRAPPER_CALL;
 
-  params = nr_php_get_zval_object_property(arg1, "params" TSRMLS_CC);
-  if (0 == params) {
-    nrl_verbosedebug(NRL_FRAMEWORK, "CakePHP: no params found in request");
+  request = nr_php_call(this_var, "getRequest");
```

Comment:
`\Cake\Controller\Controller` class has `$request` property. Why not avoid calling PHP code via `nr_php_call` and use the same method that was used to obtain the controller name: i.e. `nr_php_get_zval_object_property(this_var, "request");`?

Reply:
Using `nr_php_get_zval_object_property(this_var, "request");` does not work for CakePHP 3.x because the `$request` property is deprecated in version `3.6.0`: https://github.com/cakephp/cakephp/blob/571fb867995ccfa4034ec5b66577f0ab2acd4429/src/Controller/Controller.php#L126. It also mentions to use `getRequest` instead.

Reply:
`$name` property is also protected in [3.x](https://github.com/cakephp/cakephp/blob/3.x/src/Controller/Controller.php#L104), [4.x](https://github.com/cakephp/cakephp/blob/4.x/src/Controller/Controller.php#L118) and [5.x](https://github.com/cakephp/cakephp/blob/5.x/src/Controller/Controller.php#L114) and using `nr_php_get_zval_object_property(this_var, "name");` works so why it doesn't work for `$request`? The deprecation comment is about it not being `public` any more so PHP code will no longer be able to access `$request` property. However the agent has access to `private` and `protected` properties of `this`.

Reply:
I'm fine leaving it as is.